apiserver/facades/client/client/status.go

```diff
@@ -531,32 +538,48 @@ type statusContext struct {
 // machine and machines[1..n] are any containers (including nested ones).
 //
 // If machineIds is non-nil, only machines whose IDs are in the set are returned.
-func fetchMachines(st Backend, machineIds set.Strings) (map[string][]*state.Machine, error) {
-	v := make(map[string][]*state.Machine)
+func (context *statusContext) fetchMachines(st Backend) error {
```

Comment:
Is it worth return early for CAAS? Saves you a DB query.

Reply:
Only on CAAS models, but yes, can do this.